daemon/qrexec-daemon.c

```diff
@@ -1203,6 +1247,7 @@ _Noreturn static void handle_execute_service_child(
             };
             poll(fds, 1, -1);
             size_t l;
+terminate:
```

Comment:
Do not jump into some nested control structure. https://xkcd.com/292/
I sometimes tolerate goto to do cleanups at the end of the function (since C has no try/finally or similar constructs), but do not abuse it.
(and in fact I think some jumps into here are wrong, may kill non-disposable qube too).